.github/workflows/maven.yaml

```diff
@@ -0,0 +1,37 @@
+# Adapted from the Github Actions template `ci/maven`
+
+# This workflow will build a Java project with Maven, and cache/restore any dependencies to improve the workflow execution time
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-java-with-maven
+
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+name: Java CI with Maven
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  build-and-test:
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          java-version: '17'
+          distribution: 'corretto'
+          cache: maven
+      - name: Build and Test with Maven
+        run: mvn --batch-mode --file pom.xml package
```

Comment:
nit: it is not necessary to specify `pom.xml` location

Reply:
If default (and this is what all other developers would expect) works, it is better to use default. It is also more common to use `-B` instead of `--batch-mode` and `-D` instead of `--define`.